Objects/typeobject.c

```diff
@@ -966,6 +960,9 @@ type_get_annotations(PyTypeObject *type, void *context)
                         } else {
                             Py_DECREF(annotations);
                         }
+                    } else {
```

Comment:
This makes it so an exception during execution of co_annotations doesn't get eaten silently